### DISCUSSION TITLE
dual module issue

### DISCUSSION THREAD

**Discussion by Elllin** [pmndrs/jotai#2856] [Q&A]
On the server, jotai is being loaded in different formats:

In workspaces/<app>, jotai is resolved as CJS
In workspaces/shared, jotai is resolved as ESM
This inconsistency causes separate store instances to be created on the server, so when the app hydrates, changes are not reflected in shared

### Environment
Next.js: v12
Node.js: v18
Project Module Format: CommonJS (CJS)

How can I ensure that libraries like jotai are loaded consistently as either ESM or CJS in the server environment to maintain a single store instance...??!

**Comment by dai-shi** [comment]
Our decision was that we don't solve dual module hazard on the library end.

So, basically, you have to make sure to use CJS only in your project.
Alternatively, if the issue is only with the default store, a workaround would be to create store explicitly.
```jsx
import { createStore, Provider } from 'jotai';

const store = createStore();

  // in root component
  <Provider store={store}>

```
